library/cwakuv2.nim

```diff
@@ -0,0 +1,160 @@
+  # Convers an error message into an error-JsonResponse
+  # {
+  #   error: string;
+  # }
+  return $(%* { "error": message })
+
+################################################################################
+### Exported components
+
+proc show_version(): cstring {.dynlib, exportc.} =
+  return wakuNode2VersionString
+
+proc waku_set_event_callback(callback: EventCallback) {.dynlib, exportc.} =
+  eventCallback = callback
+
+proc waku_content_topic(appName: cstring,
+                        appVersion: uint,
+                        contentTopicName: cstring,
+                        encoding: cstring,
+                        outContentTopic: var string) {.dynlib, exportc.} =
+  # https://rfc.vac.dev/spec/36/#extern-char-waku_content_topicchar-applicationname-unsigned-int-applicationversion-char-contenttopicname-char-encoding
+  outContentTopic = fmt"{appName}/{appVersion}/{contentTopicName}/{encoding}"
+
+proc waku_pubsub_topic(topicName: cstring, encoding: cstring, outPubsubTopic: var string) {.dynlib, exportc.} =
+  # https://rfc.vac.dev/spec/36/#extern-char-waku_pubsub_topicchar-name-char-encoding
+  outPubsubTopic = fmt"/waku/2/{topicName}/{encoding}"
+
+proc waku_default_pubsub_topic(defPubsubTopic: var string) {.dynlib, exportc.} =
+  # https://rfc.vac.dev/spec/36/#extern-char-waku_default_pubsub_topic
+  defPubsubTopic = DefaultPubsubTopic
+
+proc waku_relay_publish(pubSubTopic: cstring,
+                        jsonWakuMessage: cstring,
+                        timeoutMs: int,
+                        jsonResp: var string)
+
+                        {.dynlib, exportc, cdecl.} =
+  # https://rfc.vac.dev/spec/36/#extern-char-waku_relay_publishchar-messagejson-char-pubsubtopic-int-timeoutms
+
+  var jsonContent:JsonNode
+  try:
+    jsonContent = parseJson($jsonWakuMessage)
+  except JsonParsingError:
+    jsonResp = errResp (fmt"Problem parsing json message. {getCurrentExceptionMsg()}")
+    return
+
+  var wakuMessage: WakuMessage
+  try:
+    var version = 0'u32
+    if jsonContent.hasKey("version"):
+      version = (uint32) jsonContent["version"].getInt()
+
+    wakuMessage = WakuMessage(
+        # Visit https://rfc.vac.dev/spec/14/ for further details
+        payload: jsonContent["payload"].getStr().toSeq().mapIt(byte (it)),
+        contentTopic: $jsonContent["content_topic"].getStr(),
+        version: version,
+        timestamp: getTime().toUnix(),
+        ephemeral: false
+    )
+  except KeyError:
+    jsonResp = errResp(fmt"Problem building the WakuMessage. {getCurrentExceptionMsg()}")
+    return
+
+  let targetPubSubTopic = if $pubSubTopic == "":
+                            cstring(DefaultPubsubTopic)
+                          else:
+                            pubSubTopic
+
+  let pubMsgFut = wakunode2.publishMessage(targetPubSubTopic, wakuMessage)
+
+  # With the next loop we convert an asynchronous call into a synchronous one
+  for i in 0 .. timeoutMs:
+    if pubMsgFut.finished():
+      break
+    sleep(1)
```

Comment:
`chronos` has a `withTimeout` function that i think could be used here instead of `sleep`:
https://github.com/status-im/nim-chronos/blob/master/chronos/asyncloop.nim#L1116

Reply:
Super interesting @richard-ramos. Thanks for that!
However, we cannot invoke `await`, and therefore, return a Future[..] in this proc because it is exported to C and this call needs to be synchronous.